### PR TITLE
new(falco): add json_include_message_property to have a message field without date and priority

### LINKED ISSUE #2985: Change the format of the 'output' field in the JSON payload
**Motivation**

Right now, the default output for Falco is stdout with `basic text` as format. The generated log lines follow this pattern `<timestamp> <Priority> <output>`:

```
14:37:27.505989596: Warning Detected ptrace PTRACE_ATTACH attempt (proc_pcmdline=%proc.pcmdline evt_type=%evt.type user=%user.name user_uid=%user.uid user_loginuid=%user.loginuid process=%proc.name proc_exepath=%proc.exepath parent=%proc.pname command=%proc.cmdline terminal=%proc.tty exe_flags=%evt.arg.flags %container.info)
```

By choosing the JSON format we get the same content in the `output` field, with same elements `timestamp` and `Priority`, but these elements are also contained in specific fields of the JSON:

```json
{
    "output": "14:37:27.505989596: Warning Detected ptrace PTRACE_ATTACH attempt (proc_pcmdline=%proc.pcmdline evt_type=%evt.type user=%user.name user_uid=%user.uid user_loginuid=%user.loginuid process=%proc.name proc_exepath=%proc.exepath parent=%proc.pname command=%proc.cmdline terminal=%proc.tty exe_flags=%evt.arg.flags %container.info)",
    "priority": "WARNING",
    "rule": "PTRACE attached to process",
    "time": "2023-12-20T14:37:27.505989596Z",
    "output_fields": {
        "container.info": "container.info",
        "evt.arg.flags": "evt.arg.flags",
        "evt.type": "evt.type",
        "proc.cmdline": "proc.cmdline",
        "proc.exepath": "proc.exepath",
        "proc.name": "proc.name",
        "proc.pcmdline": "proc.pcmdline",
        "proc.pname": "proc.pname",
        "proc.tty": "proc.tty",
        "user.loginuid": "user.loginuid",
        "user.name": "user.name",
        "user.uid": "user.uid"
    },
    "hostname": "host-7.local",
    "source": "syscalls",
    "tags": [
        "maturity_stable",
        "host",
        "container",
        "process",
        "mitre_privilege_escalation",
        "T1055.008"
    ]
}
```

It implies we have a duplication of the information between the fields, and it creates a mess for systems trying to deduplicate alerts. See [this issue for Falcosidekick](https://github.com/falcosecurity/falcosidekick/issues/711).

**Feature**

I propose to allow an opt-in to remove the timestamp and/or the priority from the `outpout` when the JSON format is used.

**Alternatives**

I created a PR for Falcosidekick allowing to reformat the output field. See [PR#729](https://github.com/falcosecurity/falcosidekick/pull/729)

**Additional context**

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area engine

**What this PR does / why we need it**:

There are multiple discussions (https://github.com/falcosecurity/falco/issues/2985) about allowing JSON output that has a field containing only the rule output. One proposal is to change the output format but that would cause breaking changes, and also several tools (including possibly Falcosidekick) rely on a specific output format so it seems too risky to do that. However, we still want the pure unfiltered rule output, so we're introducing a new field called `"message"` to do just that. You can turn on and off both the `output` and `message` fields from your `falco.yaml`.

Example of the difference between the two:
```
  "message": "A process has started! (proc_exe=cat proc_sname=bash)",
```

```
  "output": "15:47:29.809967681: Warning A process has started! (proc_exe=cat proc_sname=bash)",
```

NOTE1: This PR is WIP because it depends on: https://github.com/falcosecurity/falco/pull/3308 , but it's useful to run the CI
NOTE2: I also wrote a much heavier modification to the test suite to allow this to be tested, but it involves creating sinsp events and so it's a bit messy, it'll require more discussion

**Which issue(s) this PR fixes**:

Fixes #2985

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(falco): add json_include_message_property to have a message field without date and priority
```

### CODE REVIEW

**Comment by LucaGuerra** [comment]
false positive

/unhold

File: falco.yaml [hunk 1]
```diff
@@ -518,6 +518,15 @@ json_output: false
...
 json_include_output_property: true
 
+# [Incubating] `json_include_message_property`
+#
+# When using JSON output in Falco, you have the option to include the formatted
+# rule output without timestamp or priority. For instance, if a rule specifies
+# an "output" property like "Opened process %proc.name" the "message" field will
+# only contain "Opened process bash" whereas the "output" field will contain more
+# information.
+json_include_message_property: false
```
Comment (LucaGuerra) [inline falco.yaml:528]: cc @Issif proposed UX

Comment (Issif) [reply falco.yaml:528]: ok for me, it's a good start

**Review by FedeDP — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]

**Pull request merged**